src/main/scala/com/fulcrumgenomics/util/Metric.scala

```diff
@@ -70,17 +70,17 @@ object Metric {
     * line should be a single metric. */
   def read[T <: Metric](lines: Iterator[String], source: Option[String] = None)(implicit tt: ru.TypeTag[T]): Seq[T] = {
     if (lines.isEmpty) throw new IllegalArgumentException(s"No header found in metrics" + source.map(" in source: " + _).getOrElse(""))
-    val names = lines.next().split(Delimiter)
-    if (lines.isEmpty) Seq.empty
-    else {
-      lines.zipWithIndex.map { case (line, lineNumber) =>
-        val values = line.split(Delimiter)
-        if (names.length != values.length) {
-          throw new IllegalArgumentException(s"Line ${lineNumber+1} did not have the same length (${values.length}) as the header (${names.length}")
-        }
-        build[T](names.zip(values))
-      }.toSeq
-    }
+    val parser = new DelimitedDataParser(lines=lines, delimiter=Delimiter, ignoreBlankLines=false, trimFields=true)
+    val names  = parser.headers
+    parser.zipWithIndex.map { case (row, lineNumber) =>
+      val values = names.indices.map { i => row[String](i) }.map { value =>
```

Comment:
I think it's probably totally reasonable to add some methods to DelimitedDataParser to access the `String` values more efficiently.  I would just like to do it in a way that prevents the caller from modifying the underlying values.  Two possibilities:

1) Provide a `Row.field(name: String): String` which does it more efficiently.  I'm struggling to come up with a good name for the method though.
2) Provide a `toMap: Map[String,String]` that just puts all the non-None values into a map by field name and returns them?